### PR TITLE
Prevent exception when encountering a node of unknown type

### PR DESCRIPTION
**Pull request by nicolaslt**
This naively decides iteration over all properties is preferable to throwing an exception. Some more info in [this `estraverse` bug](https://github.com/estools/estraverse/issues/32).

I'm starting to think it could be better to expose the `fallback` option to `esquery` users so `esquery` users could decide what they prefer for themselves?

### CODE REVIEW

**Comment by nicolaslt** [comment]
This PR is soon going to have been open for a year without any comments or feedback, I'm tempted to close it. Any reactons?

File: tests/unknownNodeType.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    "jstestr/test",
+    "./fixtures/unknownNodeTypeAST",
+], function (esquery, assert, test, AST) {
+
+    test.defineSuite("Unknown node type", {
+
+        "does not throw": function () {
+            var thrown = false
+            try {
+              esquery(AST, '.*');
```
Comment (michaelficarra) [inline tests/unknownNodeType.js]: This selector is invalid. You probably meant to use `*`?

File: tests/unknownNodeType.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+            } catch (e) {
+              thrown = true
```
Comment (michaelficarra) [inline tests/unknownNodeType.js]: Just `assert.fail()` here.

**Comment by nicolaslt** [comment]
Moved to #78

**Pull request closed without merging**